rr_evgp/src/costmap_plugins/global_center_path_layer.cpp

```diff
@@ -0,0 +1,235 @@
+                unsigned int wallLengthPixels = master_grid.cellDistance(wall_length_);
+                unsigned int wallDistanceBehindStartPixels = master_grid.cellDistance(wall_distance_behind_start_);
+                unsigned int wallThicknessPixels = master_grid.cellDistance(wall_thickness_);
+                int mx;
+                int my;
+                master_grid.worldToMapEnforceBounds(init_robot_x, init_robot_y, mx, my);
+                cv::Point2d wallMidpoint =
+                      offsetPoint(cv::Point(mx, my), init_robot_yaw, -1.0 * wallDistanceBehindStartPixels / 2.0);
+                cv::Point2d wallPtA = offsetPoint(wallMidpoint, init_robot_yaw + CV_PI / 2.0, wallLengthPixels / 2.0);
+                cv::Point2d wallPtB = offsetPoint(wallMidpoint, init_robot_yaw - CV_PI / 2.0, wallLengthPixels / 2.0);
+
+                unsigned int goalDistanceBehindStartPixels = master_grid.cellDistance(goal_distance_behind_start_);
+                cv::Point2d goalPt =
+                      offsetPoint(cv::Point(mx, my), init_robot_yaw, -1.0 * goalDistanceBehindStartPixels);
+
+                int cx;
+                int cy;
+                master_grid.worldToMapEnforceBounds(curr_robot_x, curr_robot_y, cx, cy);
+
+                cv::Mat debug;
+                cv::cvtColor(mat_grid, debug, CV_GRAY2BGR);
+                cv::line(debug, wallPtA, wallPtB, CV_RGB(255, 0, 0),
+                         wallThicknessPixels);  // add wall in between start and end points
+
+                cv::Mat channels[3];
+                cv::split(debug, channels);
+
+                cv::Mat obstacleGrid;
+                cv::resize(channels[2], obstacleGrid,
+                           cv::Size(master_grid.getSizeInCellsX() / scaling_factor_,
+                                    master_grid.getSizeInCellsY() / scaling_factor_));
+                cv::threshold(obstacleGrid, obstacleGrid, 127, UniformCostSearch::OBSTACLE_SPACE,
+                              cv::THRESH_BINARY);  // threshold after scaling because of interpolation (halfway at 127
+                                                   // seems fine)
+
+                // Ensures track walls are at least 2 pixels wide (can't sneak through) and adds buffer in case of badly
+                // sensed wall
+                cv::Mat kernel = cv::getStructuringElement(cv::MORPH_CROSS, cv::Size(3, 3));
+                cv::dilate(obstacleGrid, obstacleGrid, kernel);
+
+                cv::bitwise_not(obstacleGrid, obstacleGrid);  // Flip due to how DistanceTransform defines obstacles
+                cv::Mat distanceGrid;
+                cv::distanceTransform(obstacleGrid, distanceGrid, cv::DIST_L2, cv::DIST_MASK_3);
+                // convert based on max so that we minimize the center path
+                double min;
+                double max;
+                cv::minMaxLoc(distanceGrid, &min, &max);
+                cv::absdiff(distanceGrid, cv::Scalar(max), distanceGrid);
```

Comment:
Could do `cv::add` with `-max` or just do `cv::absdiff`. Not sure why there isn't a 'cv::diff` function